validator/validate.js

```diff
@@ -0,0 +1,470 @@
+#!/usr/bin/env node
+
+var fs = require("fs");
+nconf = require('nconf');
+util = require('util');
+request = require('request');
+glob = require('glob');
+var Ajv  = require('ajv');
+
+const path = require('path');
+
+//
+// Setup nconf to use (in-order):
+//   1. Command-line arguments
+//   2. Environment variables
+//   3. 'config.json' file
+//   4. defaults
+nconf.argv({
+  "i": {
+    alias: 'fiware:importSchemas',
+    describe: 'Additional schemas that will be included during validation. Default imported schemas are: common-schema.json, geometry-schema.json',
+    type: 'array'
+  },
+  "w" : {
+    alias: 'fiware:warnings',
+    describe: 'How to handle FIWARE Data Models checks warnings.\n true (default) - print warnings, but does not fail. \n ignore -  do nothing and do not print warnings.\n fail - print warnings, and fails.',
+    type: 'string'
+  },
+  "p" : {
+    alias: 'fiware:path',
+    describe: 'The path of FIWARE Data Model(s) to be validated (if recursion enabled, it will be the starting point of recursion)',
+    demand: false,
+    type: 'string'
+  },
+  "h" : {
+    alias: 'help',
+    describe: 'Print the help message',
+    demand: false
+  }
+},"Usage: validate -p DataModel -w ignore -i [common-schema.json,geometry-schema.json]").file('config.json');
+
+if (nconf.get('fiware:importSchemas') == null) {
+  nconf.set('fiware:importSchemas', ['common-schema.json','geometry-schema.json']);
+}
+if (nconf.get('fiware:warnings') == null) {
+  nconf.set('fiware:warnings', 'true');
+}
+if (nconf.get('fiware:warningChecks') == null) {
+  nconf.set('fiware:warningChecks', ['schemaExist','docExist','docFolderExist','exampleExist','modelNameValid','readmeExist']);
+}
+if (nconf.get('fiware:recursiveScan') == null) {
+  nconf.set('fiware:recursiveScan', true);
+}
+if (nconf.get('fiware:validateExamples') == null) {
+  nconf.set('fiware:validateExamples', true);
+}
+if (nconf.get('fiware:loadModelCommonSchemas') == null) {
+  nconf.set('fiware:loadModelCommonSchemas', true);
+}
+if (nconf.get('fiware:importExternalSchemaFolders') == null) {
+  nconf.set('fiware:importExternalSchemaFolders', true);
+}
+if (nconf.get('fiware:resolveRemoteSchemas') == null) {
+  nconf.set('fiware:resolveRemoteSchemas', false);
+}
+if (nconf.get('fiware:ignoreFolders') == null) {
+  nconf.set('fiware:ignoreFolders', ['harvest','auxiliary']);
+}
+if (nconf.get('fiware:docFolders') == null) {
+  nconf.set('fiware:docFolders', ['doc']);
+}
+if (nconf.get('fiware:externalSchemaFolders') == null) {
+  nconf.set('fiware:externalSchemaFolders', ['externalSchema']);
+}
+if (nconf.get('ajv:missingRefs') == null) {
+  nconf.set('ajv:missingRefs', 'true');
+}
+if (nconf.get('ajv:extendRefs') == null) {
+  nconf.set('ajv:extendRefs', 'fail');
+}
+if (nconf.get('ajv:allErrors') == null) {
+  nconf.set('ajv:allErrors', true);
+}
+
```

Comment:
this block of code could be resolved more elegantly by iterating over a hash

Reply:
it's a bug fix, due to the fact that the nconf.default of the library (nconf being a sort of hash), is not working properly.

Reply:
then you can move this code out to another module so that it can be properly removed when the bug from nconf is fixed. Another option would be to overwrite the nconf functions, all in all this is just Javascript

thanks!